File: src/main/java/com/paragon/client/systems/module/hud/impl/Ping.java
package com.paragon.client.systems.module.hud.impl;

import com.paragon.client.systems.module.hud.HUDModule;
import com.paragon.client.systems.module.impl.client.Colours;
import net.minecraft.util.text.TextFormatting;

public class Ping extends HUDModule {

    public static Ping INSTANCE;

    public Ping() {
        super("Ping", "Displays your ping in ms");

        INSTANCE = this;
    }

    @Override
    public void render() {
        renderText("Ping: " + TextFormatting.WHITE + getPing() + "ms", getX(), getY(), Colours.mainColour.getValue().getRGB());
    }

    @Override
    public float getWidth() {
        return getStringWidth("Ping: " + getPing() + "ms");
    }

    @Override
    public float getHeight() {
        return getFontHeight();
    }

    public int getPing() {
        if (mc.getConnection() != null && mc.getConnection().getPlayerInfo(mc.getSession().getProfile().getId()) != null) {
            return mc.getConnection().getPlayerInfo(mc.getSession().getProfile().getId()).getResponseTime();
        }

        return -1;
    }
}

File: src/main/java/com/paragon/client/systems/module/hud/impl/Ping.kt
@file:Suppress("SuspiciousVarProperty")

package com.paragon.client.systems.module.hud.impl

import com.paragon.api.util.render.font.FontUtil
import com.paragon.client.systems.module.hud.HUDModule
import com.paragon.client.systems.module.impl.client.Colours
import net.minecraft.util.text.TextFormatting

/**
 * @author Surge
 */
object Ping : HUDModule("Ping", "Displays your ping in ms") {

    override fun render() {
        FontUtil.drawStringWithShadow("Ping: " + TextFormatting.WHITE + getPing() + "ms", x, y, Colours.mainColour.value.rgb)
    }

    override var width = FontUtil.getStringWidth("Ping: " + getPing() + "ms")
        get() = FontUtil.getStringWidth("Ping: " + getPing() + "ms")

    override var height = FontUtil.getHeight()
        get() = FontUtil.getHeight()

    fun getPing(): Int =
        if (minecraft.connection != null && minecraft.connection!!.getPlayerInfo(minecraft.player.uniqueID) != null) {
            minecraft.connection!!.getPlayerInfo(minecraft.player.uniqueID).responseTime
        } else {
            -1
        }


}